src/api/entities/NumberedPortfolio.ts

```diff
@@ -75,9 +81,17 @@ export class NumberedPortfolio extends Portfolio {
       },
       context,
     } = this;
-
-    const rawPortfolioName = await portfolio.portfolios(did, numberToU64(id, context));
-
+    const [rawPortfolioName, exists] = await Promise.all([
+      portfolio.portfolios(did, numberToU64(id, context)),
+      this.exists(),
+    ]);
+
+    if (!exists) {
```

Comment:
Another approach would be to check the name being being non-empty.

It would save us a call, but duplicates the logic. Since its in parallel I think consistency is better.